river/compose/product.py

```diff
@@ -0,0 +1,73 @@
+import itertools
+
+from river import utils
+
+from . import union
+
+__all__ = ["TransformerProduct"]
+
+
+class TransformerProduct(union.TransformerUnion):
+    """Computes interactions between the outputs of a set transformers.
+
+    This is for when you want to add interaction terms between groups of features. It may also
+    be used an alternative to `feature_extraction.PolynomialExtender` when the latter is overkill.
```

Comment:
should be changed to PolynomialFeatures

Reply:
Actually I'm keeping the name `PolynomialExtender`